hdr_html_canvas_element.md

```diff
@@ -0,0 +1,744 @@
+* [0, 12], which we will call non-normalized scene light space.
+* The [BT.2408 Recommendation](https://www.itu.int/dms_pub/itu-r/opb/rep/R-REP-BT.2408-3-2019-PDF-E.pdf) is that an HLG signal value of 0.75 map to diffuse white, which we have previously defined to be the scene light value of 1.
+  * If we are to apply no OOTFs, then this would mean scaling the interval to be [0, 1/0.265], because 0.265 is the pre-image of 0.75 under the HLG OETF.
+  * If we were to bend the definition of scene light and apply the semi-relevant 1000-nit-max OOTF, then the interval would be [0, 1/0.203].
+
+The third choice is to be discarded because it is a descent into chaos.
+If an application desires to specify a CSS style that coincides with an HLG signal of 0.75, the application is free to do a bit of math and specify the pre-image of the HLG signal of 0.75, previously noted above to be ``color(srgb-linear 0.265, 0.265, 0.265)``.
+
+There are two good reasons to prefer normalized scene light to non-normalized scene light.
+
+The first reason to prefer normalized scene light is that the numbers are round and predictable.
+
+The second reason to prefer normalized scene light is that the non-normalized representation will not behave well if applied to a non-SDR canvas. In particular, it will saturate and start to clamp at a signal value of 0.5.
+A defining property of HLG is the smooth transition between SDR and HDR, this clamping is not a smooth transition, and so something different would need to be done in non-SDR canvases.
+That would end up with a different scene light representation of an HLG image depending on whether it is destined for an HDR or SDR canvas, which is a mess from the perspective of the application and the browser.
+
+### Mode 0: SDR
+
+For completeness, there does exist one more HDR mode, the no-HDR HDR mode (which is the current default behavior).
+
+In this mode, scene light is clamped to the range [0, 1], and the SDR OOTF is applied.
+This is done by having the native application (the browser in this case) provide the operating system with a buffer in a specified color space.
+The EOTF (the inverse-OETF and then the OOTF) is then applied by some combination of the operating system, graphics hardware, and display device.
+
+For HLG images, the behavior that falls out is for the OOTF to not be applied.
+The HLG OOTF is the identity function if the display maximum luminance is 334 nits, so, for HLG images, the behavior for non-HDR canvases will be to display them as though the target display were 334 nits, which is a reasonable maximum luminance value to assume an SDR monitor has.
+
+For PQ images, the behavior that falls out is for the images to be clamped beyond 100 nits.
+
+## Proposed API outline
+
+To enable the configuration of HDR compositing.
+
+```idl
+  partial interface HTMLCanvasElement {
+    void configureHDR(HTMLCanvasCompositingMode mode, optional HTMLCanvasHDRMetadata metadata);
```

Comment:
Should this be specified for OffscreenCanvas as well? Presumably that would affect both ImageBitmap objects created via `transferToImageBitmap`, and if the OffscreenCanvas was controlling an HTMLCanvasElement, then would also affect the canvas, similarly to how resizing an OffscreenCanvas eventually causes the HTMLCanvasElement to resize.

Reply:
Suppose we have a canvas that has had an image drawn into it (SDR, PQ, or HLG) and nothing else. The goal that I have in mind is for the result of transferToImageBitmap to have the same contents as an ImageBitmap created from the original image would. Because of how everything is defined in "device independent light" (in the new upload), this ends up being the case -- no compositing mode information needed(!).

The only place where this could fall apart is if the ImageBitmap can be assigned to an <img> tag. In that case, the resulting <img> would not look the same as the canvas that it came from. Is that an option?

Reply:
> Suppose we have a canvas that has had an image drawn into it (SDR, PQ, or HLG) and nothing else. The goal that I have in mind is for the result of transferToImageBitmap to have the same contents as an ImageBitmap created from the original image would. Because of how everything is defined in "device independent light" (in the new upload), this ends up being the case -- no compositing mode information needed(!).
> 
> The only place where this could fall apart is if the ImageBitmap can be assigned to an  tag. In that case, the resulting  would not look the same as the canvas that it came from. Is that an option?

(Sorry, which tag? &lt;img&gt;? I think Github's editor deleted the angle-bracketed-tag name you referenced.)

HTMLImageElement's src attribute can not be set to an ImageBitmap. ImageBitmap in general can only be used with the imperative JavaScript APIs like the canvas rendering contexts, OffscreenCanvas, etc.